src/react/projects/spark-core-react/src/SprkDropdown/SprkDropdown.js

```diff
@@ -0,0 +1,262 @@
+/* global window */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { uniqueId } from 'lodash';
+import SprkIcon from '../SprkIcon/SprkIcon';
+
+class SprkDropdown extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      triggerText: props.defaultTriggerText,
+      isOpen: false,
+      choiceItems: props.choices.items.map(
+        item => ({ id: uniqueId(), ...item }),
+      ),
+    };
+    this.toggleDropdownOpen = this.toggleDropdownOpen.bind(this);
+    this.closeOnEsc = this.closeOnEsc.bind(this);
+    this.closeOnClickOutside = this.closeOnClickOutside.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.selectChoice = this.selectChoice.bind(this);
+    this.setSelectedChoice = this.setSelectedChoice.bind(this);
+    this.myRef = React.createRef();
```

Comment:
Looks like this one is still called myRef. It could be more descriptive of what the reference is but I won't block the PR for it.